src/vos/vos_dtx.c

```diff
@@ -1506,6 +1285,47 @@ vos_dtx_register_record(struct umem_instance *umm, umem_off_t record,
 	return rc;
 }
 
+int
+vos_dtx_register_ilog(struct umem_instance *umm, umem_off_t record,
+		      umem_off_t *tx_id)
+{
+	struct dtx_handle		*dth = vos_dth_get();
+	struct vos_dtx_entry_df		*dtx;
+	struct vos_dtx_record_df	*rec;
+	umem_off_t			 rec_umoff = UMOFF_NULL;
+	int				 rc = 0;
+
+	if (dth == NULL) {
+		*tx_id = UMOFF_NULL;
+		return 0;
+	}
+
+	rec_umoff = umem_zalloc(umm, sizeof(struct vos_dtx_record_df));
+	if (dtx_is_null(rec_umoff))
+		return -DER_NOSPACE;
+
+	rec = umem_off2ptr(umm, rec_umoff);
+	rec->tr_type = DTX_RT_ILOG;
+	rec->tr_flags = 0;
+	rec->tr_record = record;
+	rec->tr_next = UMOFF_NULL;
+
+	if (dtx_is_null(dth->dth_ent)) {
+		rc = vos_dtx_alloc(umm, dth, rec_umoff, &dtx);
+		if (rc != 0)
+			return rc;
+	} else {
+		vos_dtx_append(umm, dth, rec_umoff, record, DTX_RT_ILOG, 0,
+			       &dtx);
+	}
+
+	/* Incarnation log entry implies a share */
+	dtx->te_flags |= DTX_EF_SHARES;
```

Comment:
I admit I don't understand this code very well.   I set it so that key modifications would get committed quickly.   Without this setting, I get tons of retries on fetch.

Reply:
For the shared case, the original logic will use sync commit, but for non-shared case, it will be batched commit asynchronous.